mesop/server/server.py

```diff
@@ -106,7 +109,23 @@ def generate_data(ui_request: pb.UiRequest) -> Generator[str, None, None]:
         yield from yield_errors(runtime().get_loading_errors()[0])
 
       if ui_request.HasField("init"):
-        yield from render_loop(path=ui_request.path, init_request=True)
+        page_config = runtime().get_page_config(path=ui_request.path)
+        if page_config and page_config.on_load:
+          result = page_config.on_load(LoadEvent(path=ui_request.path))
+          # on_load is a generator function then we need to iterate through
+          # the generator object.
+          if result:
+            for _ in result:
+              yield from render_loop(
+                path=ui_request.path, init_request=True, keep_alive=True
+              )
+              runtime().context().set_previous_node_from_current_node()
+              runtime().context().reset_current_node()
```

Comment:
[question] Did you test cases where component diffing could come into play? I feel like that's an area for potential regression. Most likely should be fine. But a good place to have some test coverage.

Reply:
I do a test across navigations here: mesop/tests/e2e/on_load_test.ts

Reply:
Found an existing regression and filed: https://github.com/google/mesop/issues/371